src/appReview/saga.ts

```diff
@@ -0,0 +1,18 @@
+import InAppReview from 'react-native-in-app-review'
+import { call, takeLatest } from 'redux-saga/effects'
+import { Actions, initializeAppReview, updateAppReview } from 'src/appReview/actions'
+import { safely } from 'src/utils/safely'
+
+export function* initAppReview() {
+  yield call(initializeAppReview)
+}
+
+export function* setAppReview() {
+  const result = yield call(InAppReview.RequestInAppReview)
+  yield call(updateAppReview, result)
+}
+
+export function* appReviewSaga() {
+  yield takeLatest(Actions.INITIALIZE_APP_REVIEW, safely(initAppReview))
+  yield takeLatest(Actions.UPDATE_APP_REVIEW, safely(setAppReview))
+}
```

Comment:
maybe i am missing something here. i thought the logic would be something like on app start, spawn a single saga that checks the previous time the user has been prompted in redux. if the time is null or above a threshold, display the review popup and update the prompted time in redux.

i guess i'm confused about this `appReviewSaga` and the actions. i was expecting just one saga that checks this, and one action for updating the timestamp (whether the user rated or not). why do we need  UPDATE_APP_REVIEW and INITIALIZE_APP_REVIEW?

Reply:
> logic would be something like on app start, spawn a single saga that checks the previous time the user has been prompted in redux.

I could have this wrong, but we wait to initialize the app review saga until the user has completed onboarding.
Then after the user has completed a specified event, such as sending a transaction, we want to dispatch the app rating. This should happen only if the device supports app rating, the user hasn't rated previously, and they haven't seen the rating in n amount of time.

Reply:
So one fun thing I learned in implementing this is promise `InAppReview.RequestInAppReview()` doesn't actually return if the use rated or not. Just the fact that it was displayed.